ophyd/PreDefinedPositions.py

```diff
@@ -0,0 +1,321 @@
+        if no path found otherwise returns a list of 'locations' that define the path. If to_location 
+        is None then it returns a dictionary showing the shortest path to all possible locations. If
+        from_location is None it returns a dictionary showing the shortest path from all locations to
+        the current location. If both are None it returns a dictioanry of dictionaries. If 
+        from_location is 'current_location' the starting point is changed to the current location.
+        Parameters
+        ---------
+        from_location: string
+            The name of the starting location required for the path.
+        to_location: string
+            The name of the ending location required for the path.
+        path_list: list, output
+            A list locations indicating the path to take to reach the required position.
+        '''
+
+
+        if from_location is not 'current_location':
+            path_list = nx.shortest_path(self.nxGraph,source=from_location,target=to_location)
+        elif isinstance(self.status_list,str) and self.neighbours is not None:
+            path_list=(self.status_list)
+        else:
+            if self.neighbours is None:
+                path_list=[to_location]
+            else:
+                path_list=[]
+                for location in self.status_list:
+                    prev_path_list=path_list
+                    path_list=nx.shortest_path(self.nxGraph,source=location,target=to_location)
+
+                    if len(prev_path_list)>1 and len(prev_path_len)<len(path_list):
+                        path_list=prev_path_list
+
+        return path_list
+
+
+    def visualize_paths(self, fig_size = [10,10], axis_labels = ['arb. axis', 'arb. axis'] ,
+                        options = {}):
+        ''' Creates a plot of the possible paths between the predefined locations.
+
+        PARAMETERS
+        ----------
+        fig_size: list, optional
+            A list containing the horizontal and vertical size to make the figure.
+        axis_labels: list, optional 
+            A list of horizontal and vertical axis names for the figure.
+        options: dict, optional
+            An optional dictionary that contains kwargs for the draw_networkx function from the
+            networkx module.
+
+        '''
+
+        if self.locations is None:
+            print ('No locations to visualize')
+        else:
+            
+            default_options={'pos':nx.circular_layout(self.nxGraph),'node_color':'darkturquoise',
+                    'edge_color':'grey','node_size':6000, 'width':3,'arrowstyle':'-|>',
+                    'arrow_size':12}
+            default_options.update(options)
+
+            plt.figure('visualize {} paths'.format(self.name),figsize=(fig_size[0],fig_size[1]))
+            plt.xlabel(axis_labels[0])
+            plt.ylabel(axis_labels[1])
+            nx.draw_networkx(self.nxGraph,arrows=True,**default_options)
```

Comment:
Please format the code above better, so that the `default_options` dict would look more readable and the blank spaces after `,`'s are added where necessary.